### DISCUSSION TITLE
Hiding components vs disabling them vs Enabling with error messages

### DISCUSSION THREAD

**Discussion by pguptaDS** [twilio-labs/paste#1090] [Q&A] [answered]
I was wondering if the Paste team has any opinions on hiding components vs disabling them depending on a user's access? For example, if your role in console doesn't give you access to a certain functionality, should we hide the component all together or disable it with a tooltip or enable it and allow the user to click on it and then throw an error message?

**Comment by SiTaggart** [comment, marked as answer]
Hi @pguptaDS 

My first initial thought for this would be that it is largely going to be a decision you make based on testing. I don't necessarily believe there is a one size fits all approach. 

Given that is it based on access or privilege rights, a user may never get to use those controls, so if you hide them they will continue to live out a peaceful and blissfully ignorant life never knowing those controls exist. 

Equally, if they do some day get to play with them, or it's an important task, knowledge of their existence might be useful. Consider a "super important critical task", it might be useful in a specific context to know that an action can be taken, regardless of my inability to do so, as I can use that information to escalate it to an appropriate person who _can_ and _needs_ to take action, especially if it's time sensitive.

So test it and let us know what you find.

Some concrete advice I can give is: 

- We do have some guidance on disabled controls under [Button](https://paste.twilio.design/components/button#disabled) and [Input](https://paste.twilio.design/components/input/#validation) that largely revolves around making sure people aren't guessing why this thing is disabled.

- I've previously covered [communicating the disabled states of buttons](https://github.com/twilio-labs/paste/discussions/765#discussioncomment-80009) before and the same advice and guidance applies to all interactive controls.

Happy testing! 🎉
